scipy/linalg/special_matrices.py

```diff
@@ -498,22 +498,33 @@ def block_diag(*arrs):
     If all the input arrays are square, the output is known as a
     block diagonal matrix.
 
-    Empty sequences (i.e., array-likes of zero size) are ignored.
+    Empty sequences (i.e., array-likes of zero size) will not be ignored.
+    Noteworthy, both [] and [[]] are treated as matrices with shape ``(1,0)``.
 
     Examples
     --------
     >>> from scipy.linalg import block_diag
+    >>> from numpy import zeros
     >>> A = [[1, 0],
     ...      [0, 1]]
     >>> B = [[3, 4, 5],
     ...      [6, 7, 8]]
     >>> C = [[7]]
+    >>> P = zeros((2, 0), dtype='int32')
```

Comment:
I will not change this line, this line is in the docstring, `np` is not imported yet.
And I see this kind of usage elsewhere in the same file(line 457) so it will not ruin the consistency.